### PR TITLE
refactor: Make use-pollable-resource exhaustive deps fix

### PR DESCRIPTION
**Pull request by gorandalum**
- Adding prevState to dependency array of reload funcion is safe as
  it is a reference.
- Adding setisLoading to dependency array of reload function is
  safe as it is referentially stable.
- Adding reload to dependency array of the reload-on-focus
  useEffect is safe because the original
  load-on-reload-function-change useEffect now sets firstLoadIsDone
  to false.
- AbortController changed to ref as we don't want to add it in the
  dependency list for the useInterval hook.

And as a final note we should probably look into using useReactQuery
for this logic, as it supports most of what usePollableResource is
trying to achieve.

### Acceptance criteria
- [ ] Trip pattern updates should work as before: Should update every 20 seconds, not update when not focused, and always update when getting focus.

### CODE REVIEW

**Review by marius-at-atb — APPROVED** [review]

File: src/utils/use-previous.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import {useEffect, useRef} from 'react';
 
+/**
+ * Returns a current reference to the previous value. The returned value is safe
+ * to use in dependency arrays as refs won't trigger reruns.
+ */
```
Comment (marius-at-atb) [inline src/utils/use-previous.ts:6]: Good to have comments like this imo :+1:

Comment (mikaelbr) [reply src/utils/use-previous.ts:6]: Agreed. Easy to skip, but jsdoc/tsdoc like comments (as included on `usePollableResource`) is very handy to understand context for parameters and intention of function as well. So could even add parameters here and description of return value 😄

**Review by mikaelbr — APPROVED** [review]
Haven't seen any technically wrong things, but kind of hard to say for this hook I think. It's fairly complex. Suppor the idea of either using something like react query or swr type of semantics at some point which also relies more heavily on caching and mutations.

File: src/utils/use-pollable-resource.ts [hunk 2]
```diff
@@ -75,26 +75,27 @@ export const usePollableResource = <T, E extends Error = Error>(
...
         });
     },
-    [callback],
+    [callback, prevState, setIsLoading],
   );
 
   useEffect(() => {
     const abortController = new AbortController();
-    setAbortController(abortController);
+    firstLoadIsDone.current = false;
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/utils/use-pollable-resource.ts:83]: I have tested a bit locally, and it seems fine. 

The ref is only used in the `useEffect` which reloads on focus, and I think the reason is so reloads doesn't happen until the initial query is done.

I think it is more correct with the change, as earlier if a screen was focused and callback function changed at the same time, both the initial query and focus reload would be run. With the change now the focus reload won't happen on callback change.

**Pull request merged**